tests/test_horizon.py

```diff
@@ -85,4 +90,20 @@
     with pytest.raises(asyncio.TimeoutError):
         raise handler.exception()
 
-# TODO: test horizon retries once we move to response mocking
\ No newline at end of file
+
+@pytest.mark.asyncio
+async def test_horizon_retry(setup):
+
+    async with Horizon(setup.horizon_endpoint_uri) as horizon:
+        horizon._session.get = MagicMock(side_effect=ClientConnectionError)
+        start = time.time()
+        with pytest.raises(HorizonRequestError):
+            await horizon.account('GA3FLH3EVYHZUHTPQZU63JPX7ECJQL2XZFCMALPCLFYMSYC4JKVLAJWM')
+
+        elapsed = time.time() - start
+        # Sum of arithmetic progression
+        a_n = horizon.num_retries - 1
+        expected_time = ((2 * horizon.backoff_factor + (a_n - 1) * horizon.backoff_factor) *
```

Comment:
not sure I understand why it's calculated dynamically, in this case, there should be a single retry? if that the case then hardcode the expected time instead of calculating it (cause if for instance `horizon. num_retries` is lying - test will pass)
for completeness, I would add a test for 5 retry failure with the expectation of total failure. (test the edge cases)